lib/index.js

```diff
@@ -8,6 +8,7 @@ var Spawn = require("child_process").spawn
   , Ul = require("ul")
   , OneByOne = require("one-by-one")
   , Sliced = require("sliced")
+  , Q = require('q')
```

Comment:
I would recommend against having both promises and a callback. If you just keep the callback option, there are extended promise libraries (like Q) that can "promisify" the function.